### PR TITLE
content: Handle image emoji

### PR DESCRIPTION
**Pull request by chrisbobbe**
image

### CODE REVIEW

File: lib/widgets/content.dart [hunk 1]
```diff
@@ -432,20 +432,34 @@ class MessageUnicodeEmoji extends StatelessWidget {
...
+        clipBehavior: Clip.none,
+        children: [
+          const SizedBox(width: size, height: kBaseFontSize),
+          Positioned(
+              // Web's css makes this seem like it should be -0.5, but that looks
+              // too low.
+              top: -1.5,
+              child: Image.network(
+                adjustedSrc.toString(),
+                filterQuality: FilterQuality.low,
```
Comment (gnprice) [inline lib/widgets/content.dart:458]: This should be `FilterQuality.medium`, same as for avatars and attached images.

At `low`, it… actually it looks pretty much fine in your screenshot from iOS, but here's what it is for me on Linux desktop:
image
Pretty pixelated and jagged.

At `medium`, it looks much better, and pretty similar to what web gets in Chrome on Linux:
image
(left is Flutter desktop, right is Chrome)

… OK, and trying it on Android, on a physical Pixel 5, both `low` and `medium` look good; in fact I can't tell them apart, at least not on this emoji at this size. It seems possible that on that device they actually end up selecting the exact same algorithm.

In your testing, do you find `low` is actually better than `medium`? If so, perhaps we should choose different algorithms on different platforms.

In any case we almost surely want the same `filterQuality` value here as in sender avatars, and likely also in attached images. So let's keep it in sync with those.

**Review by gnprice — COMMENTED** [review]
Thanks @chrisbobbe for taking care of this! This looks good; small comment below.

One commit-message nit:
```
$ git uvr --name-only
8089a9551 msglist [nfc]: Explicitly model realm- and zulip-extra emoji the same way
lib/model/content.dart
lib/widgets/content.dart
83005b5ef widgets [nfc]: Make kBaseFontSize a double, instead of inferred int
lib/widgets/content.dart
15d3ba952 content: Handle image emoji
lib/model/content.dart
lib/widgets/content.dart
```

I'd give all three of these the prefix `content`.

In particular I've been keeping `message_list` for the sorts of things that go in the `message_list.dart` files — basically, everything about the message lists except the contents of the messages. (The contents are a bigger area of work here than they are in the RN app, because we're not using a webview and handing the HTML and CSS to a browser to interpret.)

File: lib/widgets/content.dart (hunk 1 above)
Comment (chrisbobbe) [reply lib/widgets/content.dart:458]: > In your testing, do you find `low` is actually better than `medium`? If so, perhaps we should choose different algorithms on different platforms.

No, I just figured that emojis are probably close to the smallest images one would want to render, and so `low` would be appropriate. Seeing that it looks worse than `medium` on some platforms, and that we don't have anything close to a performance problem yet, I'm happy to use `medium`. 🙂

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by chrisbobbe** [comment]
Hmm, I *did* push my revision to my branch, `pr-handle-image-emoji`; the tip is https://github.com/chrisbobbe/zulip-flutter/commit/5f87cc5a90d290627afd5f98a2aa51cfdfc6f590. But I'm not seeing this PR update to reflect that.

**Comment by gnprice** [comment]
> Hmm, I _did_ push my revision to my branch, `pr-handle-image-emoji`; the tip is https://github.com/chrisbobbe/zulip-flutter/commit/5f87cc5a90d290627afd5f98a2aa51cfdfc6f590. But I'm not seeing this PR update to reflect that. Is there maybe a repo-level setting?

That page 404s for me, and I can't fetch from that repo over Git.

I suspect the issue is that your fork is private (which would have been because this repo was still private when you made the fork), while this repo is now public. I guess I'll get to see the new revision when I'm back from vacation in January :slightly_smiling_face:

**Comment by chrisbobbe** [comment]
My next revision is over at #4.

**Pull request closed without merging**